### PR TITLE
Wrap POSIX’s exec for future swift run PR

### PR DESCRIPTION
**Pull request by hartbit**

### CODE REVIEW

File: Sources/POSIX/exec.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ 
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import libc
+
+/// Replace the current process image with a new process image.
+public func exec(_ path: String, _ args: [String]) throws {
+    let nullTerminatedArgs = args.map({ $0.withCString(strdup) }) + [nil]
```
Comment (aciidb0mb3r) [inline Sources/POSIX/exec.swift]: We will need to free this. We use similar pattern in many places now, we should get a utility method for this.

**Comment by hartbit** [comment]
As `CStringArray` is now in `Basic`, I can't use it from the `POSIX` target. So I moved the code to a static function in `Process` as you once suggested.

File: Sources/Basic/Process.swift [hunk 2]
```diff
@@ -467,6 +470,18 @@ extension Process {
         return try checkNonZeroExit(arguments: args, environment: environment)
     }
 
+    /// Replace the current process image with a new process image.
+    ///
+    /// - Parameters:
+    ///   - path: The path to the process executable.
+    ///   - arguments: The process arguments.
+    static public func exec(path: String, arguments: String...) throws {
```
Comment (aciidb0mb3r) [inline Sources/Basic/Process.swift]: We should keep this API similar to `Process`'s and extract the path from first argument.

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Pull request merged**